src/installer/managed/Microsoft.NET.HostModel/AppHost/PEUtils.cs

```diff
@@ -41,6 +41,37 @@ public static bool IsPEImage(string filePath)
             }
         }
 
+        /// <summary>
+        /// Remove the CET Compat bit from extended DLL characteristics
+        /// </summary>
+        /// <param name="file">Memory-mapped PE file</param>
+        /// <param name="accessor"></param>
+        internal static void RemoveCetCompatBit(MemoryMappedFile file, MemoryMappedViewAccessor accessor)
+        {
+            using (PEReader reader = new PEReader(file.CreateViewStream(0, 0, MemoryMappedFileAccess.Read)))
+            {
+                // https://learn.microsoft.com/windows/win32/debug/pe-format#debug-type
+                const int IMAGE_DEBUG_TYPE_EX_DLLCHARACTERISTICS = 20;
+                foreach (DebugDirectoryEntry entry in reader.ReadDebugDirectory())
+                {
+                    if ((int)entry.Type != IMAGE_DEBUG_TYPE_EX_DLLCHARACTERISTICS)
```

Comment:
dont think there are multiple `IMAGE_DEBUG_TYPE_EX_DLLCHARACTERISTICS` sections right? should it break after unsetting the CET_COMPAT bit at the end?